debacl/utils.py

```diff
@@ -109,6 +113,19 @@ def knn_graph(X, k, method='brute_force', leaf_size=30):
             raise ImportError("The scikit-learn library could not be loaded." +
                               " It is required for the 'ball-tree' method.")
 
+    if method == 'precomputed':
+        # Pseudo-sort each row of the distance matrix so that the indices of the closest k elements are listed first
+        # The entries [...,:k] are not in sorted order by distance
+        neighbors = _np.argpartition(X, k)[:, 0:k]
```

Comment:
Good idea - I didn't know numpy had this!  On a 1,000 x 1,000 matrix, my crude ipython timing says this is 4x faster than argsort.